discoverd/health/register_test.go

```diff
@@ -136,7 +136,11 @@ func (RegisterSuite) TestRegister(c *C) {
 			Events:    make(chan MonitorEvent),
 		}
 		hb := reg.Register()
-		defer hb.Close()
+		defer func() {
+			go func() { <-unregisterChan }()
```

Comment:
1559e5fc620367c9e1c1c1d47c03420598feb40f calls the wrapped heartbeater's close method, which then [triggers the mocked `Close()` method](https://github.com/flynn/flynn/blob/1559e5fc620367c9e1c1c1d47c03420598feb40f/discoverd/health/register_test.go#L61), that sends a notification to the `unregisterChan`. I'm making sure the channel gets drained here, so that the function does not block indefinitely.